src/main/scala/firrtl/transforms/Flatten.scala

```diff
@@ -0,0 +1,113 @@
+// See LICENSE for license details.
+
+package firrtl
+package passes
+
+import firrtl.ir._
+import firrtl.Mappers._
+import firrtl.annotations._
+import scala.collection.mutable
+
+// Tags an annotation to be consumed by this pass
+object FlattenAnnotation {
+  def apply(target: Named): Annotation = Annotation(target, classOf[Flatten], "")
+
+  def unapply(a: Annotation): Option[Named] = a match {
+    case Annotation(named, t, _) if t == classOf[Flatten] => Some(named)
+    case _ => None
+  }
+}
+
+//  
+// Takes all annotations for module instances and inline the entire hierarchy of modules down from the given ones.
+// This transformation is based on InlineInstances transformation. More concretely, it fully reuses run method and overriding execute method to collect the right set of modules to inline 
+// Note: Inlining a module means inlining all its children module instances   
+class Flatten extends InlineInstances {
```

Comment:
Don't extend InlineInstances - you should instead extend Transform, and then instantiate a new InlineInstances within this transform, and call its run method explicitly. The Annotation system relies on matching the class of the transform with its annotation, so Flatten, when it calls getMyAnnotations, would get both Flatten and InlineInstance Annotations.